pkg/commands/os.go

```diff
@@ -139,7 +140,15 @@ func (c *OSCommand) PrepareSubProcess(cmdName string, commandArgs ...string) *ex
 
 // Quote wraps a message in platform-specific quotation marks
 func (c *OSCommand) Quote(message string) string {
+	fmt.Println(c.Platform.os)
 	message = strings.Replace(message, "`", "\\`", -1)
+	if c.Platform.os == "linux" {
+		if strings.ContainsRune(message, '\'') {
+			c.Platform.escapedQuote = `"`
```

Comment:
I don't think we should be changing c.Platform at runtime. The idea with that is to set up platform-specific constants at compile time and then testing will be easier because we will know what their values are.

Could we perhaps make this code non-platform specific by saying something like:
```
escapedQuote := c.Platform.escapedQuote
if strings.Contains(message, c.Platform.escapedQuote) {
  escapedQuote = c.Platform.fallbackEscapedQuote
}
return escapedQuote + message + escapedQuote
```
and then add a fallbackEscapedQuote value to the platform files

Reply:
Pushed a commit which is working and addresses your comments